### PR TITLE
kernel: do not strip proc by default

### PR DESCRIPTION
**Pull request by PolynomialDivision**
Currently, you are not able to get statistics about IPv4 and IPv6
usage. This information can be collected via the snmp and snmp6.
However, in the current state this interface is disabled as you can
can read in the "902-debloat_proc.patch":
 "Strip non-essential /proc functionality to reduce code size"

Tools like netstat use the snmp/6 interface to collect interface
statistics. Some prometheus exporters also mention this:
- prometheus-collectors/netstat.lua
- prometheus-collectors/snmp6 (still a PR)
- collectd/snmp6 (still a PR)

PRs:
- https://github.com/collectd/collectd/pull/3789
- https://github.com/openwrt/packages/pull/14158

Do u still think that the proc needs to be stripped by default?

### CODE REVIEW

File: target/linux/generic/config-5.4 [hunk 1]
```diff
@@ -4120,7 +4120,7 @@ CONFIG_PRINT_STACK_DEPTH=64
 CONFIG_PROC_FS=y
 # CONFIG_PROC_KCORE is not set
 # CONFIG_PROC_PAGE_MONITOR is not set
-CONFIG_PROC_STRIPPED=y
+CONFIG_PROC_STRIPPED=n
```
Comment (adschm) [inline target/linux/generic/config-5.4:4123]: Wrong syntax (just check the rest of the file ...)

**Comment by adschm** [comment]
>  "Strip non-essential /proc functionality to reduce code size"

So, please guess what's the obvious question here, which is not answered in the commit message ... ;-)

File: target/linux/generic/config-5.4 (hunk 1 above)
Comment (PolynomialDivision) [reply target/linux/generic/config-5.4]: whuuups. sry. changed.

**Comment by PolynomialDivision** [comment]
> > "Strip non-essential /proc functionality to reduce code size"
> 
> So, please guess what's the obvious question here, which is not answered in the commit message ... ;-)

I will make some measurements regarding image size. ;)

**Comment by PolynomialDivision** [comment]
Non-Stripped:
```
binwalk openwrt-ath79-generic-tplink_tl-wdr4300-v1-squashfs-sysupgrade.bin 

DECIMAL       HEXADECIMAL     DESCRIPTION
--------------------------------------------------------------------------------
512           0x200           LZMA compressed data, properties: 0x6D, dictionary size: 8388608 bytes, uncompressed size: 6447691 bytes
2015404       0x1EC0AC        Squashfs filesystem, little endian, version 4.0, compression:xz, size: 2737138 bytes, 864 inodes, blocksize: 262144 bytes, created: 2020-12-09 21:07:25
```
```
Stripped:
binwalk openwrt-ath79-generic-tplink_tl-wdr4300-v1-squashfs-sysupgrade.bin 

DECIMAL       HEXADECIMAL     DESCRIPTION
--------------------------------------------------------------------------------
512           0x200           LZMA compressed data, properties: 0x6D, dictionary size: 8388608 bytes, uncompressed size: 6389451 bytes
1999012       0x1E80A4        Squashfs filesystem, little endian, version 4.0, compression:xz, size: 2737206 bytes, 864 inodes, blocksize: 262144 bytes, created: 2020-12-09 23:56:43
```

So it is 0.05824 MB.or in other words the image grows 0.9%.

**Comment by adschm** [comment]
> So it is 0.05824 MB.or in other words the image grows 0.9%.

Nice try. That's 56.9 kiB, which would be a lot, but that's uncompressed as it appears.
I'm not sure whether I can calculate compressed size from the first column, but if that's true we would still have 16 kiB.

**Comment by PolynomialDivision** [comment]
> > So it is 0.05824 MB.or in other words the image grows 0.9%.
> 
> Nice try. That's 56.9 kiB, which would be a lot, but that's uncompressed as it appears.
> I'm not sure whether I can calculate compressed size from the first column, but if that's true we would still have 16 kiB.

Yep, but this is only true for devices with less ram and less flash, or? And we talk here about network hardware. And I would argue that ip statistics are not "non-essential". ;)

**Comment by ynezz** [comment]
Make it conditional for `default y if !SMALL_FLASH`

**Comment by PolynomialDivision** [comment]
> Make it conditional for `default y if !SMALL_FLASH`

You mean the other way around, or? Or is that ! not a negation?
`default y if SMALL_FLASH`

**Comment by PolynomialDivision** [comment]
@ynezz Sry, I'm new to this kernel config stuff. I have to remove the line `# CONFIG_PROC_STRIPPED is not set``completely, or? Otherwise the default behavior is overwritten?

I searched for examples, but I this is confusing for me.
Here is the flag set to y
https://github.com/openwrt/openwrt/blob/master/target/linux/ath25/config-5.4#L77

and here it has a depedency:
https://github.com/openwrt/openwrt/blob/master/target/linux/ath25/patches-5.4/107-ar5312_gpio.patch#L59-L65

But isn't it always now set to y?

**Comment by memberst1** [comment]
I found solution that fits the characteristics of the existing users of openwrt.
https://dev.archive.openwrt.org/ticket/15781

File: target/linux/generic/hack-5.4/902-debloat_proc.patch [hunk 2]
```diff
@@ -45,7 +45,7 @@ Signed-off-by: Felix Fietkau <nbd@nbd.name>
  	  this interface. For example, rkt is such a piece of software.
  
 +config PROC_STRIPPED
-+	default n
```
Comment (ynezz) [inline target/linux/generic/hack-5.4/902-debloat_proc.patch:48]: This is wrong, take a look for example here:

https://github.com/openwrt/openwrt/blob/866307285445f1086af1cc8352a46e55260274d1/config/Config-kernel.in#L34-L36

Comment (PolynomialDivision) [reply target/linux/generic/hack-5.4/902-debloat_proc.patch:48]: I deleted now the `depends on EXPERT` and moved the config part to `config/Config-kernel.in`

Comment (PolynomialDivision) [reply target/linux/generic/hack-5.4/902-debloat_proc.patch:48]: Something is wrong.

Comment (PolynomialDivision) [reply target/linux/generic/hack-5.4/902-debloat_proc.patch]: Here is the SWAP config option in kernel. It also has a default value?
https://github.com/torvalds/linux/blob/v5.4/init/Kconfig#L308-L316

So I leaf the proc-patch as it is and I only remove that from config-5.4 and expose it to openwrt buildchain via config/Config-kernel.in?

I think it is now how it should be.

**Review by dangowrt — APPROVED** [review]

**Comment by ynezz** [comment]
Thanks! Pulled into my staging tree at https://git.openwrt.org/openwrt/staging/ynezz.git

**Pull request closed without merging**